applicationset/generators/generator_spec_processor.go

```diff
@@ -151,11 +151,26 @@ func mergeGeneratorTemplate(g Generator, requestedGenerator *argoprojiov1alpha1.
 // "params" parameter is an array, where each index corresponds to a generator. Each index contains a map w/ that generator's parameters.
 func InterpolateGenerator(requestedGenerator *argoprojiov1alpha1.ApplicationSetGenerator, params map[string]any, useGoTemplate bool, goTemplateOptions []string) (argoprojiov1alpha1.ApplicationSetGenerator, error) {
 	render := utils.Render{}
+
+	var interpolatedGenerator *argoprojiov1alpha1.ApplicationSetGenerator
+
+	if requestedGenerator != nil && requestedGenerator.Matrix != nil {
+		// If 2nd child is a matrix, only interpolate the 1st child of the inner matrix
```

Comment:
Just thinking, this might better be...
```suggestion
		// If child is a matrix, only interpolate the 1st child of the inner matrix
```